src/geoserver/util.py

```diff
@@ -1038,6 +1038,10 @@ def wms_direct(wms_url, xml=None, version=None, headers=None):
         if exc.response.status_code == 404:
             return None
         raise exc
+    except Error as exc:
+        if exc.code == 2 and exc.data.get('http_code') == 404:
```

Comment:
Try to import
- `WebMapService` through wms.WebMapService
- WebFeatureService through wfs.WebFeatureService
- and remove other changes in this commit and in previous three commits